onnxruntime/core/session/standalone_op_invoker.cc

```diff
@@ -288,20 +371,24 @@ onnxruntime::Status CreateOp(const OrtKernelInfo* info,
                                                ep->Type(),
                                                &kernel_create_info);
   ORT_RETURN_IF_ERROR(status);
-  onnxruntime::Node node;
+  std::vector<onnxruntime::NodeArg*> input_args;
+  std::vector<onnxruntime::NodeArg*> output_args;
+  NodeHolder node_holder(new onnxruntime::Node(std::string("standalone_") + op_name, op_name, "", input_args, output_args, nullptr, domain), ReleaseNode);
   for (int i = 0; i < attr_count; ++i) {
     auto attr_proto = reinterpret_cast<const ONNX_NAMESPACE::AttributeProto*>(attr_values[i]);
-    node.AddAttributeProto(*attr_proto);
+    node_holder->AddAttributeProto(*attr_proto);
   }
+
   auto kernel_def_builder = KernelDefBuilder::Create();
   kernel_def_builder->SetName(op_name);
   kernel_def_builder->SetDomain(domain);
   kernel_def_builder->SinceVersion(version);
-  OpKernelInfo instant_kernel_info(node, *kernel_def_builder->Build(), *ep, {}, {}, {});
+  OpKernelInfo instant_kernel_info(*node_holder.get(), *kernel_def_builder->Build(), *ep, {}, {}, {});
   std::unique_ptr<onnxruntime::OpKernel> op_kernel;
   FuncManager func_mgr;
   status = kernel_create_info->kernel_create_func(func_mgr, instant_kernel_info, op_kernel);
   ORT_RETURN_IF_ERROR(status);
+  AddNode(op_kernel.get(), node_holder);
```

Comment:
This name is confusing. Would've been better to define a local struct in this file that encapsulated the addition, deletion of nodes in the map with a mutex owned by that struct. If you don't have to update the counts, you'll probably not even need all this.

Reply:
Defined a singleton class NodeRepo for node maintenance with thread safety.